pkgs/tools/inputmethods/fcitx5/with-addons.nix

```diff
@@ -1,17 +1,20 @@
-{ symlinkJoin, makeWrapper, fcitx5, fcitx5-lua, fcitx5-configtool, fcitx5-qt, fcitx5-gtk, addons ? [] }:
+{ lib, symlinkJoin, makeWrapper, fcitx5, fcitx5-configtool, fcitx5-qt, fcitx5-gtk, addons ? [ ] }:
+
+with lib;
```

Comment:
with lib should not be used globally

Reply:
> with lib should not be used globally

Is there any RFC for it or is it a convention that I am not aware of?

IMO, using `with lib;` globally is the convention because currently there are 1366[1] modules using `with lib;` and about 1497[2] modules not using it (the actual number is smaller because of the existence of generated `lib.mdDoc` with global `with lib;`).

[1]: `rg -n with\ lib\; nixos/modules | rg -e ':1:' -e ':2:'
 -e ':2:' -e ':3:' -e ':4:' -e ':5:' -e '::' -e ':6:' -e ':7:' -e ':8:' -e ':9:' | wc -l`
[2]: `rg --count-matches 'lib\.' nixos/modules | wc -l`

Reply:
> Is there any RFC for it or is it a convention that I am not aware of?

Not that I know of, supersandro has his own set of conventions. Using `with` in a too broad scope is frowned upon because it breaks tooling, can generate "subtle" errors. It's better to use `with lib;` in a minimal scope unlike here. In this peculiare instance, it could have been moved to `postBuild = with lib; ...`

Reply:
It is kinda accepted to use it in modules but it is still not great code style even there. We should limit it to scopes where it makes sense like meta.

FYI if something is used a lot in the tree does not mean it fits the current coding standard no things would be done in. Some things are just not yet changed and would only be changed when other refactors happen.